migrating.md

```diff
@@ -92,20 +97,89 @@ myApp.config(["djangoRMIProvider", function(djangoRMIProvider) {
+
+The RequireJS migration deals with both **pages** (HTML) and **modules** (JavaScript). Any individual page is either migrated or not. Individual modules are also migrated or not, but a "migrated" module may be used on both RequireJS and non-RequireJS pages.
+
+Logic in `hqModules.js` determines whether or not we're in a RequireJS environment and changes the behavior of `hqDefine` accordingly. In a RequireJS environment, `hqDefine` just passes through to RequireJS's `define`. Once all pages have been migrated, we'll be able to delete `hqModules.js` altogether and switch all of the `hqDefine` calls to `define`.
+
+These docs walk through the process of migrating a single page to RequireJS.
+
+### Basic Migration Process
+
+Prerequisites: Before a page can be migrated, **all** of its dependencies must already be in external JavaScript files and must be using `hqDefine`. See above for details on moving inline script blocks to files, and see [module patterns](https://github.com/dimagi/js-guide/blob/master/code-organization.md#module-patterns) for details on `hqDefine`. Also, pages that are not descendants of [hqwebapp/base.html](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/hqwebapp/templates/hqwebapp/base.html) cannot yet be migrated.
+
+Once these conditions are met, migrating to RequireJS is essentially the process of explicitly adding each module's dependencies to the module's definition, and also updating each HTML page to reference a single "main" module rather than including a bunch of `<script>` tags:
+1. Add `requirejs_main` tag and remove `<script>` tags
+1. Add dependencies
+1. Test
+
+Sample PRs:
+- [RequireJS migration: dashboard](https://github.com/dimagi/commcare-hq/pull/19182/) is an example of an easy migration, where all dependencies are already migrated
+- [RequireJS proof of concept](https://github.com/dimagi/commcare-hq/pull/18116) migrates a few pages (lookup tables, data dictionary) and many of our commonly-used modules (analytics, `hq.helpers.js`, etc.). This also contains the changes to `hqModules.js` that make `hqDefine` support both migrated and unmigrated pages.
+
+#### Add `requirejs_main` tag and remove `<script>` tags
+
+The `requirejs_main` tag is what indicates that a page should use RequireJS. The page should have one "main" module. Most of our pages are already set up like this: they might include a bunch of scripts, but there's one in particular that handles the event handlers, bindings, etc. that are specific to that page.
+
+It's fine for multiple pages to use the same main module - this may make sense for closely related pages.
+
+Add `{% requirejs_main 'myApp/js/myModule' %}` near the top of the template: it can go after `load` and `extends` but should appear before content blocks. Note that it's a module name, not a file name, so it doesn't include `.js`.
+
+Remove and other `<script>` tags from the file. You'll be adding these as dependencies to the main module.
```

Comment:
nit: should say "Remove other `<script>` tags...`"?

Reply:
https://github.com/dimagi/js-guide/pull/27

Thanks for reading through this :)